src/data-model-providers/endpoint/SpanEndpointProvider.h

```diff
@@ -0,0 +1,100 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/data-model-provider/MetadataTypes.h>
+#include <app/server-cluster/ServerClusterInterface.h>
+#include <data-model-providers/endpoint/EndpointProviderInterface.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/support/Span.h>
+
+#include <utility> // For std::pair
+
+namespace chip {
+namespace app {
+
+class SpanEndpointProvider : public EndpointProviderInterface
+{
+public:
+    class Builder
+    {
+    public:
+        explicit Builder(EndpointId id);
+
+        Builder & SetComposition(DataModel::EndpointCompositionPattern composition);
+        Builder & SetParentId(EndpointId parentId);
+        Builder & SetServerClusters(Span<ServerClusterInterface *> serverClusters);
```

Comment:
Need to describe ownership/lifetime for these here.

Or generally for all Span arguments.